addon/model.js

```diff
@@ -171,7 +255,7 @@ export default class MegamorphicModel extends EmberObject {
       // TODO need to walk the chain down as well to notify changes
     }
     if (this !== this._topModel) {
-      this._topModel._updateCurrentState(state);
+      this._topModel._updateCurrentState(!CUSTOM_MODEL_CLASS && state);
```

Comment:
Never, it’ll be stripped entirely if false. Also in the false branch this method takes no args

Reply:
If I run
```
EMBER_DATA_FEATURE_OVERRIDE=CUSTOM_MODEL_CLASS,REQUEST_SERVICE,IDENTIFIRS,RECORD_DATA_STATE,RECORD_
DATA_ERRORS node $(volta which ember) s --no-live-reload --port=0

I see `_updateCurrentState` called with `false` for test `3867eb4c` (`unit/model/state: isDirty`).

Also when we fix this so that we can can toggle these tests in dev mode the code won't be stripped at all, which is required for the runtime toggling to work.

```

Reply:
Calling it with false is safe (and also the current behavior elsewhere, I simply fixed a few places it had been forgotten)

Reply:
The obvious thing is that when the flag is false there is no currentState at all since we don't rely on InternalModel. We won't use the arg, and won't set anything.